pysb/bng.py

```diff
@@ -236,7 +243,7 @@ def read_simulation_results(self):
         # Read concentrations data
         cdat_arr = numpy.loadtxt(self.base_filename + '.cdat', skiprows=1)
         # Read groups data
-        if len(self.model.observables):
+        if self.model and len(self.model.observables):
```

Comment:
You're right, observables are not read in if `model=None`. The use case for `model=None` is limited (probably only internal unit tests?), but in the future best practice would probably be to check the header of the observables file and verify those match the model. Expressions should be read in from there too, to save re-computing them in PySB.